### ISSUE TITLE
glob.glob() ordering is non-deterministic across OSes, breaking split reproducibility

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#13]
### Summary
`SetCardDataModule.setup()` builds the image list using `glob.glob()`, which returns files in OS-dependent (non-deterministic) order. Although `train_test_split(..., random_state=42)` is used, the same seed only guarantees reproducibility if the input list is identically ordered — which is not the case across different filesystems or OSes (e.g. macOS HFS+ vs Linux ext4).

As a result, the train/val split is not reproducibly the same on all machines.

### Code location
`src/data/set_card_data_pipeline.py`, lines 139–149:

```python
all_images = glob.glob(os.path.join(self.data_dir, '*.jpg')) + \
             glob.glob(os.path.join(self.data_dir, '*.png'))
# ... no sort here ...
train_paths, val_paths = train_test_split(all_images, test_size=0.2, random_state=42)
```

### Fix
Sort the list before splitting:

```python
all_images = sorted(
    glob.glob(os.path.join(self.data_dir, '*.jpg')) +
    glob.glob(os.path.join(self.data_dir, '*.png'))
)
```

### Files affected
- `src/data/set_card_data_pipeline.py`
